pkg/comp-functions/functions/vshn-postgres-func/mailgun_alerting.go

```diff
@@ -0,0 +1,127 @@
+	}
+
+	if comp.Spec.Parameters.Monitoring.Email != "" {
+		log.Info("Deploying AlertmanagerConfig for mail alerting...")
+		err := deployAlertmanagerConfig(ctx, comp, iof)
+		if err != nil {
+			fmt.Println(err)
+			return runtime.NewFatalErr(ctx, "Can't deploy AlertmanagerConfig "+comp.Name+"-alertmanagerconfig-mailgun for mail alerting", err)
+		}
+
+	}
+
+	log.Info("Finishing mailgun-alerting function with NewNormal")
+
+	return runtime.NewNormal()
+}
+
+func deployAlertmanagerConfig(ctx context.Context, comp *vshnv1.VSHNPostgreSQL, iof *runtime.Runtime) error {
+	var alertManagerConfigName = comp.Name + "-alertmanagerconfig-mailgun"
+	var alertManagerConfigSecretName = comp.Name + "-alertmanagerconfig-mailgun-secret"
+	receiverName := "mailgun"
+
+	ac := &alertmanagerv1alpha1.AlertmanagerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      alertManagerConfigName,
+			Namespace: comp.Status.InstanceNamespace,
+			Labels: map[string]string{
+				"alert-manager-config": receiverName,
+			},
+		},
+		Spec: alertmanagerv1alpha1.AlertmanagerConfigSpec{
+			Receivers: []alertmanagerv1alpha1.Receiver{
+				{
+					Name: receiverName,
+					EmailConfigs: []alertmanagerv1alpha1.EmailConfig{
+						{
+							To:           comp.Spec.Parameters.Monitoring.Email,
+							From:         "appcat@appuio.cloud",
+							AuthUsername: "appcat@appuio.cloud",
+							AuthPassword: &v1.SecretKeySelector{
+								Key: "password",
+								LocalObjectReference: v1.LocalObjectReference{
+									Name: alertManagerConfigSecretName,
+								},
+							},
+							Smarthost:    "smtp.eu.mailgun.org:465",
+							RequireTLS:   pointer.Bool(true),
+							SendResolved: pointer.Bool(true),
+						},
+					},
+				},
+			},
+			Route: &alertmanagerv1alpha1.Route{
+				GroupBy: []string{
+					"alertname",
+				},
+				GroupWait:      "10s",
+				GroupInterval:  "5m",
+				RepeatInterval: "1h",
+				Receiver:       receiverName,
+			},
+		},
+	}
+
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      alertManagerConfigSecretName,
+			Namespace: comp.Status.InstanceNamespace,
+		},
+	}
+
+	xRef := xkube.Reference{
+		DependsOn: &xkube.DependsOn{
+			APIVersion: "v1",
+			Kind:       "Secret",
+			Namespace:  comp.Status.InstanceNamespace,
+			Name:       alertManagerConfigSecretName,
+		},
+	}
+
+	patchSecretWithOtherSecret := xkube.Reference{
+		PatchesFrom: &xkube.PatchesFrom{
+			DependsOn: xkube.DependsOn{
+				APIVersion: "v1",
+				Kind:       "Secret",
+				Namespace:  iof.Config.Data["emailAlertingSecretNamespace"],
+				Name:       iof.Config.Data["emailAlertingSecretName"],
+			},
+			FieldPath: pointer.String("data.password"),
+		},
+		ToFieldPath: pointer.String("data.password"),
+	}
+
+	if err := iof.Desired.PutIntoObject(ctx, secret, comp.Name+"-alertmanagerconfig-mailgun-secret", patchSecretWithOtherSecret); err != nil {
+		return err
+	}
+
+	return iof.Desired.PutIntoObject(ctx, ac, comp.Name+"-alertmanagerconfig-mailgun", xRef)
```

Comment:
Nitpick: use `alertManagerConfigName` and `alertManagerConfigSecretName` you already defined above.